packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -319,19 +313,30 @@ abstract class Scheduler extends BindingBase {
   /// device's screen is turned off it will typically be delayed until
   /// the screen is on and the application is visible.)
   void ensureVisualUpdate() {
-    if (_hasRequestedABeginFrameCallback)
+    if (_hasScheduledFrame || _inFrame)
+      return;
+    _scheduleFrame();
+  }
+
+  void _scheduleFrame() {
+    if (_hasScheduledFrame)
       return;
     ui.window.scheduleFrame();
-    _hasRequestedABeginFrameCallback = true;
+    _hasScheduledFrame = true;
   }
 
   /// Whether the scheduler is currently handling a "begin frame"
   /// callback.
   ///
   /// True while [handleBeginFrame] is running in checked mode. False
   /// otherwise.
-  static bool get debugInFrame => _debugInFrame;
-  static bool _debugInFrame = false;
```

Comment:
(in the latter case, explain why it's not exposed and what bugs would be likely in the dartdocs)